documenters_aggregator/spiders/det_schools.py

```diff
@@ -0,0 +1,152 @@
+# -*- coding: utf-8 -*-
+import scrapy
+import re
+import urllib.parse
+from datetime import datetime
+from documenters_aggregator.spider import Spider
+
+
+class Det_schoolsSpider(Spider):
+    name = 'det_schools'
+    long_name = 'Detroit Board of Education'
+    allowed_domains = ['detroitk12.org']
+    start_urls = ['http://detroitk12.org/board/meetings/']
+
+    def parse(self, response):
+        """
+        `parse` should always `yield` a dict that follows the Event Schema
+        <https://city-bureau.github.io/city-scrapers/06_event_schema.html>.
+
+        Change the `_parse_id`, `_parse_name`, etc methods to fit your scraping
+        needs.
+        """
+
+        names = response.xpath('//main/h3/a/text()').extract()
+        calendar_links = response.xpath('//main/h3/a/@href').extract()
+
+        times = response.xpath('//main/h3/following-sibling::text()[1]').extract()
+        times = map(lambda x: x.strip(), times)
+
+        addresses = response.xpath('//main/h3/following-sibling::span[@class="address"]/text()').extract()
+
+        items = zip(names, calendar_links, times, addresses)
+
+        for item in items:
+            data = {
+                '_type': 'event',
+                'id': self._parse_id(item[1]),
+                'name': item[0],
+                'description': item[0],
+                'classification': self._parse_classification(item),
+                'start_time': self._parse_start(item[2]),
+                'end_time': self._parse_end(item[2]),
+                'timezone': 'America/Detroit',
+                'status': self._parse_status(item),
+                'all_day': self._parse_all_day(item),
+                'location': self._parse_location(item[3]),
+                'sources': self._parse_sources(item),
+            }
+
+            data['id'] = self._generate_id(data, data['start_time'])
+
+            yield data
+
+    def _parse_next(self, response):
+        """
+        Get next page. You must add logic to `next_url` and
+        return a scrapy request.
+        """
+        next_url = None  # What is next URL?
+        return scrapy.Request(next_url, callback=self.parse)
+
+    def _parse_id(self, item):
+        parsed_url = urllib.parse.urlparse(item)
+        parsed_query = urllib.parse.parse_qs(parsed_url.query)
+        return parsed_query['eid'][0]
```

Comment:
I am using the Google Cal event ID as the unique ID string. They look like `MmlhcTcyNW50Nm43dWZqN3BpOWwzYmF1ZzRfMjAxODA1MTVUMTMwMDAwWiA4bmpua21kbzgxcDdyZGw0MjA4dDI2MmM2b0Bn`. Any problems with the length?

Reply:
Hmm, interesting. That is a really long ID. It's ugly but shouldn't break anything... 🤞